pyproject.toml

```
@@ -33,13 +32,11 @@ requires = ["poetry_core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = "^3.8"
 protobuf = "^4.21"
-shapely = [
-    { version = "<2", python = "<3.8" },
-    { version = "*", python = ">=3.8" }
-]
+shapely = "^2.0.0"
 pyclipper = "^1.3.0"
+pyproj = "^3.4.1"
```

if we want to keep the pyproj dependency, it would be nice to relax this to being optional, with a run time check if it's used. Adding pyproj generally sounds heavy to me just to support this scenario. Even if you're on board with the transform fn idea as the specification parameter and we don't technically need pyproj, we might want to offer a convenience wrapper to facilitate the common pyproj usage. (If this doesn't make sense, I can make a pr or write the code out to help explain)

Seems like an [extra](https://python-poetry.org/docs/pyproject/#extras)  would work here, right? Something like:

```suggestion
pyproj = { version = "^3.4.1", optional = true }

[tool.poetry.extras]
proj = ["pyproj"]
```

Regardless, can we add a note to the readme for this case?